core/services/relay/evm/target_strategy.go

```diff
@@ -0,0 +1,226 @@
+package evm
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+	"sync/atomic"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/google/uuid"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+	"github.com/smartcontractkit/chainlink-evm/pkg/writetarget"
+)
+
+var (
+	_           writetarget.TargetStrategy = &evmTargetStrategy{}
+	ErrTxFailed                            = errors.New("submitted transaction failed")
+)
+
+type evmTargetStrategy struct {
+	cr commontypes.ContractReader
+	cw commontypes.ContractWriter
+
+	lggr      logger.Logger
+	forwarder string
+
+	receiverGasMinimum uint64
+	binding            commontypes.BoundContract
+	bound              atomic.Bool
+}
+
+type TransmissionInfo struct {
+	GasLimit        *big.Int
+	InvalidReceiver bool
+	State           uint8
+	Success         bool
+	TransmissionId  [32]byte
+	Transmitter     common.Address
+}
+
+// The gas cost of the forwarder contract logic, including state updates and event emission.
+// This is a rough estimate and should be updated if the forwarder contract logic changes.
+// TODO: Make this part of the on-chain capability configuration
+const ForwarderContractLogicGasCost = 100_000
+
+const (
+	TransmissionStateNotAttempted uint8 = iota
+	TransmissionStateSucceeded
+	TransmissionStateInvalidReceiver
+	TransmissionStateFailed
+)
+
+type Config struct {
+	// Address of the contract that will get the forwarded report
+	Address string
+	// Optional gas limit that overrides the default limit sent to the chain writer
+	GasLimit *uint64
+}
+
+func NewEVMTargetStrategy(cr commontypes.ContractReader, cw commontypes.ContractWriter, forwarder string, gasLimitDefault uint64, lggr logger.Logger) *evmTargetStrategy {
+	bound := atomic.Bool{}
+	bound.Store(false)
+	return &evmTargetStrategy{
+		cr:                 cr,
+		cw:                 cw,
+		lggr:               lggr,
+		forwarder:          forwarder,
+		receiverGasMinimum: gasLimitDefault - ForwarderContractLogicGasCost,
+		binding: commontypes.BoundContract{
+			Address: forwarder,
+			Name:    "forwarder",
+		},
+		bound: bound,
+	}
+}
+
+func (t *evmTargetStrategy) QueryTransmissionState(ctx context.Context, reportID uint16, request capabilities.CapabilityRequest) (*writetarget.TransmissionState, error) {
```

Comment:
nit: Why is this a pointer?

Reply:
This was basically carried over from the Aptos code, since this interface was based on that initial implementation https://github.com/smartcontractkit/chainlink-aptos/blob/bac568dab6c7c2c076eefb284b09cf4dc3215e96/relayer/write_target/write_target.go#L291-L332